src/usersyncroniser.ts

```diff
@@ -232,10 +232,10 @@ export class UserSyncroniser {
 
     public async GetUserUpdateState(discordUser: User, webhookID?: string): Promise<IUserState> {
         log.verbose(`State update requested for ${discordUser.id}`);
-        let mxidExtra;
+        let mxidExtra = "";
         if (webhookID) {
             // no need to escape as this mxid is only used to create an intent
-            mxidExtra = "_" + Util.ParseMxid(`@${webhookID}`, false).localpart;
+            mxidExtra = "_"  + webhookID;
         }
         const userState: IUserState = Object.assign({}, DEFAULT_USER_STATE, {
             id: discordUser.id,
```

Comment:
github won't let me comment a line below, but fwiw the bridge has utilities to create these users without tracking all this information. See the `getUserIdForSuffix` function (and `getIntentForSuffix`). Idea is you pass `@discord_.*` to the registration, and for the `ForSuffix` functions you give it `discordUser.id+mxidExtra`